cpp/src/feather/metadata.cc

```diff
@@ -326,11 +325,9 @@ class ColumnBuilder::Impl {
   ColumnType::type type_;
 
   // Type-specific metadata union
-  union {
```

Comment:
gcc 4.4 doesn't allow non-POD in a union (it complains that e.g. `CategoryMetadata` has a constructor)